polaris-service/src/main/java/org/apache/polaris/service/ratelimiter/OpenTelemetryClock.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.ratelimiter;
+
+import io.opentelemetry.sdk.common.Clock;
+
+/** Implementation of the opentelemetry Clock interface using the Polaris Clock interface */
+public class OpenTelemetryClock implements Clock {
+  org.apache.polaris.service.ratelimiter.Clock clock;
+
+  public OpenTelemetryClock(org.apache.polaris.service.ratelimiter.Clock clock) {
+    this.clock = clock;
+  }
+
+  @Override
```

Comment:
What is this overriding?

Reply:
It implements the opentelemetry Clock interface using the Polaris Clock interface. I updated it so when we just write `Clock` we always refer to a Polaris Clock and for the opentelemetry Clock we write out the full path, `io.opentelemetry.sdk.common.Clock`.

I got myself into this situation because after doing my initial implementation using the opentelemetry limiter, I figured we shouldn't be stuck to their interface, so the idea now is that we own our own Clock/RateLimiter interface but that means we need:
- an Polaris Clock implementation using an OpenTelemetry Clock
- an OpenTelemetry Clock implementation using a Polaris Clock
- an OpenTelemetry RateLimiter implementation, which normally takes an OpenTelemetry Clock, using a Polaris Clock

A lot of complexity would go away if we just committed to OpenTelemetry's interfaces so I'm open to discussion on that.

Reply:
I see. I think what I missed is that OpenTelemetry has **two** `Clock`s -- `io.opentelemetry.sdk.common.Clock` and `org.apache.polaris.service.ratelimiter.Clock`. This file LGTM to me given that.

I actually like the idea of having our own interfaces in front of the OpenTelemetry ones, if only for testing. It also fixes this odd two-Clocks situation